examples/scene/lasso.py

```diff
@@ -0,0 +1,138 @@
+# -*- coding: utf-8 -*-
+# vispy: gallery 30
+# -----------------------------------------------------------------------------
+# Copyright (c) Vispy Development Team. All Rights Reserved.
+# Distributed under the (new) BSD License. See LICENSE.txt for more info.
+# -----------------------------------------------------------------------------
+"""
+Vispy Lasso
+=============
+
+Demonstrate the use of lasso selection.
+
+The lasso selection is done on a 2D scatter but could be extended further by user.
+"""
+import sys
+import numpy as np
+from vispy import app, scene
+from vispy.geometry import curves
+from vispy.scene import visuals
+from matplotlib import path
```

Comment:
It's too bad that this has to depend on matplotlib. It doesn't seem to fail which suggests we have matplotlib in our CI environment. Do we/can we have a fallback of the bounding box implementation if matplotlib isn't available? Perhaps something like:

```
import warnings

try:
    from matplotlib import path
except ImportError:
    warnings.warn("Lasso example requires matplotlib for more accurate selection. Falling back to naive bounding box selection.")
    path = None
```

And then check for `None` later in the code. If you could wrap the bounding box implementation in a path-like function/class that would be even more amazing but I realize this is a big ask.